extensions/db/src/main/java/org/mule/extension/db/api/StatementStreamingResultSetCloser.java

```diff
@@ -0,0 +1,143 @@
+   * 
+   * @param connection
+   */
+  public void closeResultSets(DbConnection connection) {
+    Object connectionLock = getConnectionLock(connection);
+
+    synchronized (connectionLock) {
+      checkValidConnectionLock(connection, connectionLock);
+
+      Set<ResultSet> resultSets = connectionResultSets.get(connection);
+      if (resultSets != null) {
+        for (ResultSet resultSet : resultSets) {
+          super.close(connection, resultSet);
+        }
+      }
+
+      releaseResources(connection, connectionLock);
+    }
+  }
+
+  @Override
+  public void close(DbConnection connection, ResultSet resultSet) {
+    Object connectionLock = getTrackedConnectionLock(connection);
+
+    synchronized (connectionLock) {
+      checkValidConnectionLock(connection, connectionLock);
+
+      Set<ResultSet> resultSets = getConnectionResultSets(connection, resultSet);
+      try {
+        super.close(connection, resultSet);
+      } finally {
+        if (isEmpty(resultSets)) {
+          releaseResources(connection, connectionLock);
+        }
+      }
+    }
+  }
+
+  /**
+   * Adds a resultSet for tracking in order to be able to close it later
+   *
+   * @param connection connection that holds the resultSet
+   * @param resultSet resultSet to track
+   */
+  public void trackResultSet(DbConnection connection, ResultSet resultSet) {
+    Object connectionLock = getConnectionLock(connection);
+
+    synchronized (connectionLock) {
+      Set<ResultSet> resultSets = connectionResultSets.get(connection);
+
+      if (resultSets == null) {
+        resultSets = new HashSet<>();
+        connectionResultSets.put(connection, resultSets);
+      }
+
+      resultSets.add(resultSet);
+    }
+  }
+
+  protected Object getTrackedConnectionLock(DbConnection connection) {
+    Object connectionLock = connectionLocks.get(connection);
+
+    if (connectionLock == null) {
+      throw new IllegalStateException("Attempting to close resultSet from non tracked connection");
+    }
+
+    return connectionLock;
+  }
+
+  protected void releaseResources(DbConnection connection, Object connectionLock) {
+    connectionResultSets.remove(connection);
+    connectionLocks.remove(connectionLock);
+
+    connection.release();
+  }
+
+  protected Set<ResultSet> getConnectionResultSets(DbConnection connection, ResultSet resultSet) {
+    Set<ResultSet> resultSets = connectionResultSets.get(connection);
+
+    if (resultSets != null && !resultSets.remove(resultSet)) {
+      throw new IllegalStateException("Attempting to close non tracked resultSet");
+    }
+    return resultSets;
+  }
+
+  protected void checkValidConnectionLock(DbConnection connection, Object connectionLock) {
+    if (connectionLock != connectionLocks.get(connection)) {
+      throw new ConcurrentModificationException("Connection lock modified in another thread");
+    }
+  }
+
+  protected Object getConnectionLock(DbConnection connection) {
```

Comment:
as far as i can tell it's not possible due to the usage. Specially since this is a Kraan class from the original connector